packages/grid/x-data-grid/src/tests/sorting.DataGrid.test.tsx

```diff
@@ -44,6 +44,67 @@ describe('<DataGrid /> - Sorting', () => {
     expect(getColumnValues(0)).to.deep.equal(['10', '0', '5']);
   });
 
+  it('should allow sorting using `apiRef` for unsortable columns', () => {
+    let apiRef: React.MutableRefObject<GridApi>;
+    function TestCase() {
+      apiRef = useGridApiRef();
+      const cols = [{ field: 'id', sortable: false }];
+      const rows = [{ id: 10 }, { id: 0 }, { id: 5 }];
+
+      return (
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid apiRef={apiRef} columns={cols} rows={rows} />
+        </div>
+      );
+    }
+
+    render(<TestCase />);
+    expect(getColumnValues(0)).to.deep.equal(['10', '0', '5']);
+    const header = getColumnHeaderCell(0);
+    // should not sort on header click when `colDef.sortable` is `false`
+    fireEvent.click(header);
+    expect(getColumnValues(0)).to.deep.equal(['10', '0', '5']);
+
+    // should allow sort using `apiRef`
+    act(() => apiRef.current.sortColumn('id', 'desc'));
+    expect(getColumnValues(0)).to.deep.equal(['10', '5', '0']);
+    act(() => apiRef.current.sortColumn('id', 'asc'));
+    expect(getColumnValues(0)).to.deep.equal(['0', '5', '10']);
+    act(() => apiRef.current.sortColumn('id', null));
+    expect(getColumnValues(0)).to.deep.equal(['10', '0', '5']);
+  });
+
+  it('should allow clearing the current sorting using `sortColumn` idempotently', () => {
```

Comment:
There was a bug in sortColumn, i.e. on calling `sortColumn('field', null)`, instead of removing the current sort it circulated between ascending and descending sort.

CSB reproduction, keep pressing `None` button:
Before: https://codesandbox.io/p/sandbox/holy-field-kf2jyd?file=%2Fsrc%2FDemo.tsx%3A63%2C19
After: https://codesandbox.io/p/sandbox/jovial-bash-9kg32f?file=%2Fpackage.json%3A6%2C88